### PR TITLE
Add docs for using and creating a concept.

### PR DESCRIPTION
**Pull request by nsthorat**
Add API for concepts. Clean up ConceptInputType.

https://lilacml.com/concepts/concept_use.html
https://lilacml.com/concepts/concept_create.html

https://huggingface.co/spaces/lilacai/nikhil_staging

### CODE REVIEW

File: docs/concepts/concept_use.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+Applying a concept to text can be done in one of two ways:
+
+1. For a single document
+2. For a dataset of documents. See [Applying concepts to a Dataset](../datasets/dataset_concepts.md)
+   for how to apply a concept to a dataset.
+
+A concept is simply a collection of positive and negative examples. To use a concept, we also need
+to choose an _Embedding_. Some embeddings, like `gte` and `sbert` will run entirely on device. Other
+embeddings like `openai`, `cohere`, or `palm`. See [Concept Technical Details](concept_technical.md)
```
Comment (dsmilkov) [inline docs/concepts/concept_use.md:11]: finish sentence

File: docs/concepts/concept_use.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+## From the UI
+
+```{tip}
+[Try Lilac concepts on HuggingFace](https://huggingface.co/spaces/lilacai/lilac)
+```
+
+To use a concept from the UI, click the concept from the Navigation panel, which will open the
+concept preview page.
+
+<img src="../_static/concepts/concept_preview.png"></img>
```
Comment (dsmilkov) [inline docs/concepts/concept_use.md:26]: maybe use a more obvious example than angel eyes

File: docs/concepts/concept_create.md [hunk 3]
```diff
@@ -1 +1,101 @@
...
+### Step 2: Add positive examples
+
+Add a short concept description. The concept description can be used to seed the concept, using an
+LLM, with fake positive examples. Often times it's hard to find positive examples, or invent fake
+positive examples, so this step can be useful to seed the concept.
+
+However, it's not required. You can choose to seed the concept with as many positive examples as you
+want.
+
+c Just like in step 3, we can use an LLM to generate fake negative examples, or you can enter your
```
Comment (dsmilkov) [inline docs/concepts/concept_create.md:32]: remove "c" at start, also missing Step 3: Add negative examples header, and "just like in step 3" -> "just like in step 2"

**Review by dsmilkov — APPROVED** [review]

File: docs/concepts/concept_use.md (hunk 1 above)
Comment (nsthorat) [reply docs/concepts/concept_use.md]: Done

File: docs/concepts/concept_use.md (hunk 2 above)
Comment (nsthorat) [reply docs/concepts/concept_use.md]: Done

File: docs/concepts/concept_create.md (hunk 3 above)
Comment (nsthorat) [reply docs/concepts/concept_create.md]: Done

**Pull request merged**